### PR TITLE
[Host]-Refactor upgrade tests for Host

### PR DESCRIPTION
**Pull request by omkarkhatavkar**
```
/Users/okhatavk/satellite/robottelo/env/bin/python "/Applications/PyCharm CE.app/Contents/plugins/python-ce/helpers/pycharm/_jb_pytest_runner.py" --path /Users/okhatavk/satellite/robottelo/tests/upgrades/test_host.py -- -m pre_upgrade
Testing started at 5:17 PM ...
Launching pytest with arguments -m pre_upgrade /Users/okhatavk/satellite/robottelo/tests/upgrades/test_host.py --no-header --no-summary -q in /Users/okhatavk/satellite/robottelo

============================= test session starts ==============================
collecting ... collected 2 items / 1 deselected / 1 selected

tests/upgrades/test_host.py::TestScenarioPositiveGCEHostComputeResource::test_pre_create_gce_cr_and_host 

=========== 1 passed, 1 deselected, 2 warnings in 863.75s (0:14:23) ============

Process finished with exit code 0
2023-04-10 17:18:13 - robottelo.collection - INFO - Processing test items to add testimony token markers
PASSED [100%]


/Users/okhatavk/satellite/robottelo/env/bin/python "/Applications/PyCharm CE.app/Contents/plugins/python-ce/helpers/pycharm/_jb_pytest_runner.py" --path /Users/okhatavk/satellite/robottelo/tests/upgrades/test_host.py -- -m post_upgrade
Testing started at 5:17 PM ...
Launching pytest with arguments -m post_upgrade /Users/okhatavk/satellite/robottelo/tests/upgrades/test_host.py --no-header --no-summary -q in /Users/okhatavk/satellite/robottelo

============================= test session starts ==============================
collecting ... collected 2 items / 1 deselected / 1 selected

tests/upgrades/test_host.py::TestScenarioPositiveGCEHostComputeResource::test_post_create_gce_cr_and_host 

=========== 1 passed, 1 deselected, 2 warnings in 863.75s (0:5:23) ============

Process finished with exit code 0
2023-04-10 17:18:13 - robottelo.collection - INFO - Processing test items to add testimony token markers
PASSED [100%]


```

### CODE REVIEW

File: tests/upgrades/test_host.py [hunk 1]
```diff
@@ -16,107 +16,185 @@
...
-            architecture=arch,
-            compute_resource=gce_cr,
-            name='autoupgrade_gce_img',
-            operatingsystem=os,
-            username='gceautou',
-            uuid=LATEST_RHEL7_GCE_IMG_UUID,
-        ).create()
+        assert class_host.name == self.fullhostname
+        assert class_host.build_status_label == 'Installed'
+        assert class_host.ip == self.google_host(googleclient, None)
```
Comment (jyejare) [inline tests/upgrades/test_host.py:173]: Need not to be explicit  with `None` as its none by default in function defination.

File: tests/upgrades/test_host.py [hunk 2]
```diff
@@ -16,107 +16,185 @@
...
         save_test_data(
             {
-                'org': function_org.name,
-                'loc': loc.name,
-                'cr_name': cr_name,
+                'org': module_puppet_org.name,
+                'loc': module_puppet_loc.name,
+                'cr_name': module_gce_compute_puppet.name,
```
Comment (jyejare) [inline tests/upgrades/test_host.py:178]: Proabably you should be able to retrive all the other attributes from host object. So just keep host_name and we are good.

**Review by jyejare — CHANGES_REQUESTED** [review]
ACK pending questions/comments

File: tests/upgrades/test_host.py [hunk 3]
```diff
@@ -16,107 +16,185 @@
...
+    ):
+        """Sets Hostgroup for GCE Host Provisioning"""
+        hgroup = session_puppet_enabled_sat.api.HostGroup(
+            architecture=session_puppet_default_architecture,
+            compute_resource=module_gce_compute_puppet,
+            domain=module_puppet_domain,
+            location=[module_puppet_loc],
+            environment=module_puppet_environment,
+            puppet_proxy=session_puppet_enabled_proxy,
+            puppet_ca_proxy=session_puppet_enabled_proxy,
```
[the remark this answers is not in the document]
Comment (adarshdubey-star) [reply tests/upgrades/test_host.py]: I see @omkarkhatavkar testing it out with non-puppet sat and it should work fine

File: pytest_fixtures/component/provision_gce.py [hunk 4]
```diff
@@ -108,6 +117,16 @@ def session_puppet_default_partition_table(session_puppet_enabled_sat):
...
 
 
+@pytest.fixture(scope='session')
+def session_default_partition_table(session_target_sat):
+    ptable = (
+        session_target_sat.api.PartitionTable()
+        .search(query={'search': f'name="{DEFAULT_PTABLE}"'})[0]
+        .read()
+    )
+    return ptable
```
Comment (adarshdubey-star) [inline pytest_fixtures/component/provision_gce.py:127]: You can get rid of this one we have https://github.com/SatelliteQE/robottelo/blob/master/pytest_fixtures/component/partition_table.py#L6
```suggestion
```

File: tests/upgrades/test_host.py [hunk 5]
```diff
@@ -16,107 +16,185 @@
...
-    @pytest.fixture(scope='class')
-    def delete_host(self):
-        if self.fullhost:
-            host = entities.Host().search(query={'search': f'name={self.fullhost}'})
-            if host:
-                entities.Host(id=host[0].id).delete()
+    def google_host(self, googleclient, host_name=None):
+        """Returns the Google Client Host object to perform the assertions"""
+        if not host_name:
+            return googleclient.get_vm(name='{}'.format(self.fullhostname.replace('.', '-')))
```
[the remark this answers is not in the document]
Comment (omkarkhatavkar) [reply tests/upgrades/test_host.py]: `self.fullhostname ` has a static value that needs for assertion.

**Review by JacobCallahan — APPROVED** [review]

**Review by adarshdubey-star — APPROVED** [review]
Nicely done @omkarkhatavkar :)

**Review by jyejare — APPROVED** [review]

**Pull request merged**